src/lazy/text/buffer.rs

```diff
@@ -869,7 +930,15 @@ impl<'data> TextBufferView<'data> {
     ///
     /// If the input does not contain the entire list, returns `IonError::Incomplete(_)`.
     // TODO: DRY with `match_list`
-    pub fn match_list_1_1(self) -> IonMatchResult<'data> {
+    pub fn match_list_1_1(
+        self,
+    ) -> IonParseResult<
+        'top,
+        (
+            TextBufferView<'top>,
+            &'top [LazyRawValueExpr<'top, TextEncoding_1_1>],
+        ),
+    > {
```

Comment:
🗺️ Matching a list now returns both the matched text buffer and a bump-allocated collection of child value expressions.